ipaserver/plugins/agreement.py

```diff
@@ -0,0 +1,151 @@
+from ipalib import api, errors, output
+from ipalib import Bool, Str, StrEnum
+from ipalib.plugable import Registry
+from ipaserver.plugins.baseldap import (
+    LDAPObject, LDAPSearch, LDAPCreate, LDAPDelete,
+    LDAPUpdate, LDAPRetrieve, LDAPAddMember, LDAPRemoveMember,
+    global_output_params, pkey_to_value)
+from ipaserver.plugins.hbacrule import is_all
+from ipalib import _, ngettext
+from ipapython.dn import DN
+
+
+__doc__ = _("""
+""")
+
+register = Registry()
+
+
+@register()
+class agreement(LDAPObject):
+    """
+    User Agreement object.
+    """
+    container_dn = DN(("cn", "agreements"))
+    object_name = _('Agreement')
+    object_name_plural = _('Agreements')
+    object_class = ['ipaassociation', 'fasagreement']
+    permission_filter_objectclasses = ['fasagreement']
+    default_attributes = [
+        'cn', 'description', 'member',
+    ]
+    uuid_attribute = 'ipauniqueid'
+    attribute_members = {
+        'memberuser': ['user'],
```

Comment:
Sorry, this has to be ``'member': ['user']``. ``LDAPAddMember`` uses ``attribute_members`` to decide which type of object can be added to an attribute.